superset-frontend/src/datasource/DatasourceEditor.jsx

```diff
@@ -918,6 +919,30 @@ class DatasourceEditor extends React.PureComponent {
                   />
                 }
               />
+              <Field
+                label={t('Deprecated at')}
+                fieldKey="deprecated_at"
+                description={t('The date this metric is or will be deprecated')}
+                control={
+                  <TextControl
+                    controlId="deprecated_at"
+                    placeholder="YYYY-MM-DD"
+                  />
+                }
+              />
+              <Field
+                label={t('Deprecation link')}
+                fieldKey="deprecation_link"
+                description={t(
+                  'A link with more details regarding the deprecation of this metric',
+                )}
+                control={
+                  <TextControl
+                    controlId="deprecation_link"
+                    placeholder="https://www.example.com"
+                  />
+                }
+              />
```

Comment:
I think having a freetext description field would be more convenient for most orgs. Instead of having a dedicated link field, could we have a general description field in which one can paste links if needed?

Reply:
That's a good suggestion, how would you recommend structuring a description field like that? Ideally, we could provide the capability for someone to embed a link and make it clickable/open in a new tab. Would rendering arbitrary markdown be a bad idea?

Reply:
I think we should be ok supporting the same markdown allowances we support on the dashboard markdown component: 
![image](https://user-images.githubusercontent.com/33317356/111182123-477fcf80-85b7-11eb-982a-e1f9b1c061e8.png) 
![image](https://user-images.githubusercontent.com/33317356/111182163-51093780-85b7-11eb-8640-a76d98a05302.png)

I might be oversimplifying, but I'd be happy with just a simple freetext field that's then rendered as markdown.

Reply:
I don't think forcing a new tab in markdown is possible unless you use html `a` tags. i can play around a bit with supporting markdown descriptions. it's a bit less structured (which i'm not sure if that's good or bad).

For context, in internal use, the learn more link typically goes to a google doc or something else with lots of detailed, freeform content. So we made the choice in other products to just link out to there instead of making people write more context/details in product

Reply:
I think we can allow both the free-text markdown and the "Learn more" link, if it's not too much work.

Reply:
I found this, and making the following change seems to send the links to `_blank`: 
![image](https://user-images.githubusercontent.com/33317356/111198802-a8fc6a00-85c8-11eb-9456-65604f9a99ad.png)

```patch
diff --git a/superset-frontend/src/dashboard/components/gridComponents/Markdown.jsx b/superset-frontend/src/dashboard/components/gridComponents/Markdown.jsx
index b071a9982..5b30cd400 100644
--- a/superset-frontend/src/dashboard/components/gridComponents/Markdown.jsx
+++ b/superset-frontend/src/dashboard/components/gridComponents/Markdown.jsx
@@ -286,6 +286,7 @@ class Markdown extends React.PureComponent {
             isValidNode: node => node.type !== 'script',
           }),
         ]}
+        linkTarget="_blank"
       />
     );
   }
```